teeracle/Cargo.toml

```diff
@@ -9,130 +9,63 @@ license = "Apache-2.0"
+sp-runtime = { version = "4.1.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
+sp-std = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
 
-[dependencies.timestamp]
-default-features = false
-package = "pallet-timestamp"
-git = "https://github.com/paritytech/substrate.git"
-branch = "master"
-version = "4.0.0-dev"
-
-[dependencies.pallet-teerex]
-default-features = false
-package = "pallet-teerex"
-path = "../teerex"
-
-[dependencies.test-utils]
-default-features = false
-path = '../tests-utils'
-
-[dev-dependencies.externalities]
-package = "sp-externalities"
-git = "https://github.com/paritytech/substrate.git"
-branch = "master"
-version = "0.10.0-dev"
-
-[dev-dependencies.sp-keyring]
-git = "https://github.com/paritytech/substrate.git"
-branch = "master"
-version = "4.0.0-dev"
-
-[dev-dependencies.pallet-balances]
-git = "https://github.com/paritytech/substrate.git"
-branch = "master"
-version = "4.0.0-dev"
+# benchmarking
+frame-benchmarking = { version = "4.0.0-dev", default-features = false, optional = true, git = "https://github.com/paritytech/substrate.git", branch = "master" }
+hex-literal = { version = "0.3.2", optional = true }
+test-utils = { path = "../test-utils", optional = true, default-features = false }
+timestamp = { version = "4.0.0-dev", default-features = false, optional = true, git = "https://github.com/paritytech/substrate.git", branch = "master", package = "pallet-timestamp" }
 
-[dev-dependencies.hex-literal]
-version = "0.3.2"
+[dev-dependencies]
+externalities = { version = "0.10.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master", package = "sp-externalities" }
+frame-benchmarking = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master" }
+hex-literal = "0.3.2"
+sp-keyring = { version = "4.1.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master" }
+pallet-balances = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master" }
+test-utils = { path = "../test-utils" }
+timestamp = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master", package = "pallet-timestamp" }
 
-[dependencies.teeracle-primitives]
-default-features = false
-package = "teeracle-primitives"
-path = "../primitives/teeracle"
 
 [features]
-default = ['std']
+default = ["std"]
 std = [
-    'codec/std',
-    'log/std',
-    'frame-support/std',
-    'frame-system/std',
+    "codec/std",
+    "log/std",
+    "scale-info/std",
+    # local
     "pallet-teerex/std",
-    'scale-info/std',
-    'sp-core/std',
-    'sp-io/std',
-    'sp-runtime/std',
-    'sp-std/std',
-    'substrate-fixed/std',
     "teeracle-primitives/std",
-    # benchmarking stuff
-    "timestamp/std",
-    'frame-benchmarking/std',
-    "test-utils/std",
+    # encointer
```

Comment:
Yeah, encointer found it's way to integritee

Reply:
Upgraded substrate fixed. However, I was not able to remove `without_storage_info`, because in the teeracle pallet a String is hidden somewhere. I'd suggest fixing this in another PR, if we want to have storage_info available.

Reply:
Yes, pitty. `PalletString` is also preventing some migrations in encointer... I can strongly agree to do this in another PR, I just thought it was maybe a low-hanging fruit.

At some point this will be mandatory by substrate, so we will need to do this at some point.